Description: Removed -std=c99 compiler extra flags from numpy
 this change fixes autopkgtest failure on s390x with python-meshio
Author: Matthieu Clemenceau <matthieu.clemenceau@canonical.com>
Forwarded: no
Last-Update: 2020-05-28

--- a/setup.py
+++ b/setup.py
@@ -25,7 +25,6 @@
 import sys
 import subprocess
 import textwrap
-import sysconfig
 
 
 if sys.version_info[:2] < (3, 5):
@@ -229,40 +228,6 @@
             sdist.run(self)
 
 
-def get_build_overrides():
-    """
-    Custom build commands to add `-std=c99` to compilation
-    """
-    from numpy.distutils.command.build_clib import build_clib
-    from numpy.distutils.command.build_ext import build_ext
-
-    def _is_using_gcc(obj):
-        is_gcc = False
-        if obj.compiler.compiler_type == 'unix':
-            cc = sysconfig.get_config_var("CC")
-            if not cc:
-                cc = ""
-            compiler_name = os.path.basename(cc)
-            is_gcc = "gcc" in compiler_name
-        return is_gcc
-
-    class new_build_clib(build_clib):
-        def build_a_library(self, build_info, lib_name, libraries):
-            if _is_using_gcc(self):
-                args = build_info.get('extra_compiler_args') or []
-                args.append('-std=c99')
-                build_info['extra_compiler_args'] = args
-            build_clib.build_a_library(self, build_info, lib_name, libraries)
-
-    class new_build_ext(build_ext):
-        def build_extension(self, ext):
-            if _is_using_gcc(self):
-                if '-std=c99' not in ext.extra_compile_args:
-                    ext.extra_compile_args.append('-std=c99')
-            build_ext.build_extension(self, ext)
-    return new_build_clib, new_build_ext
-
-
 def generate_cython():
     cwd = os.path.abspath(os.path.dirname(__file__))
     print("Cythonizing sources")
@@ -425,8 +390,6 @@
             'f2py%s.%s = numpy.f2py.f2py2e:main' % sys.version_info[:2],
             ]
 
-    cmdclass={"sdist": sdist_checked,
-             }
     metadata = dict(
         name = 'numpy',
         maintainer = "NumPy Developers",
@@ -445,7 +408,8 @@
         classifiers=[_f for _f in CLASSIFIERS.split('\n') if _f],
         platforms = ["Windows", "Linux", "Solaris", "Mac OS-X", "Unix"],
         test_suite='nose.collector',
-        cmdclass=cmdclass,
+        cmdclass={"sdist": sdist_checked,
+                },
         python_requires='>=3.5',
         zip_safe=False,
         entry_points={
@@ -470,7 +434,6 @@
 
         metadata['configuration'] = configuration
         # Customize extension building
-        cmdclass['build_clib'], cmdclass['build_ext'] = get_build_overrides()
     else:
         # Version number is added to metadata inside configuration() if build
         # is run.
